src/wikigraph.cc

```diff
@@ -35,24 +40,71 @@ WikiGraph::WikiGraph(const std::string& file_name) {
   }
 }
 
-
 // --------- Algorithms ------------
 // TODO: getPathBFS()
-std::vector<std::string> WikiGraph::getPathBFS(const std::string& from_page, const std::string& to_page) const {
-  std::vector<std::string> path_to_return;
-  // ...
-  (void) from_page;
-  (void) to_page;
+std::vector<std::string> WikiGraph::getPathBFS(
+    const std::string& start_page, const std::string& end_page) const {
+  std::cout << "Finding BFS Path from [" << start_page << "] to [" <<
+    end_page << "]" << std::endl;
 
-  return path_to_return;
+  std::vector<std::string> page_path;
+
+  std::map<std::string, std::string> page_tree;
+  std::queue<std::string> to_visit;
+
+  std::string curr_page;
+  to_visit.push(start_page);
+
+  while (!to_visit.empty() && curr_page != end_page) {
+    curr_page = to_visit.front();
+    // std::cout << "curr_page: " << curr_page << std::endl;
+    to_visit.pop();
+    if (article_map.find(curr_page) != article_map.end()) {
+      // std::cout << "AH" << std::endl;
+      std::vector<std::string> adjacent_pages = article_map.at(curr_page);
+      for (const auto& adj_page : adjacent_pages) {
+        // if page hasn't been visited.
+        if (page_tree.count(adj_page) != 1) {
+          to_visit.push(adj_page);
+          page_tree[adj_page] = curr_page;
+        }
+        if (adj_page == end_page) {
+          break;  // path to page was found.
+        }
+      }
+    }
+  }
+
+  // Path between pages was not found...
+  if (page_tree.count(end_page) == 0) {
+    std::cout << "NO PATH FOUND" << std::endl;
+    return page_path;
+  }
+
+  std::cout << "Constructing Path..." << std::endl;
+
+  std::string temp_page = end_page;
+  std::vector<std::string> reverse_path;
+  while (temp_page != start_page) {
+    reverse_path.push_back(temp_page);
+    temp_page = page_tree.at(temp_page);
+  }
+  reverse_path.push_back(start_page);
+
+  for (int i = (int)reverse_path.size() - 1; i >= 0; --i) {
+    page_path.push_back(reverse_path[i]);
+  }
+
+  return page_path;
 }
 
 // TODO: getPathDijkstras()
-std::vector<std::string> WikiGraph::getPathDijkstras(const std::string& from_page, const std::string& to_page) const {
+std::vector<std::string> WikiGraph::getPathDijkstras(
```

Comment:
i prefer the start page end page terminology but you shouldn't change it in the BFS traversal (i'll implement it for the dijkstra's PR)

Reply:
i'll switch it back.